<!DOCTYPE HTML>
<html lang="en" class="sidebar-visible no-js light">
    <head>
        <!-- Book generated using mdBook -->
        <meta charset="UTF-8">
        <title>gget mutate - gget</title>


        <!-- Custom HTML head -->
        <!-- Google tag (gtag.js) -->
        <script async src="https://www.googletagmanager.com/gtag/js?id=G-ZM4N3XBEW7"></script>
        <script>
          window.dataLayer = window.dataLayer || [];
          function gtag(){dataLayer.push(arguments);}
          gtag('js', new Date());
        
          gtag('config', 'G-ZM4N3XBEW7');
        </script>

        <meta name="description" content="gget enables efficient querying of genomic reference databases">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="theme-color" content="#ffffff" />

        <link rel="shortcut icon" href="../favicon.png">
        <link rel="stylesheet" href="../css/variables.css">
        <link rel="stylesheet" href="../css/general.css">
        <link rel="stylesheet" href="../css/chrome.css">
        <link rel="stylesheet" href="../css/print.css" media="print">

        <!-- Fonts -->
        <link rel="stylesheet" href="../FontAwesome/css/font-awesome.css">
        <link rel="stylesheet" href="../fonts/fonts.css">

        <!-- Highlight.js Stylesheets -->
        <link rel="stylesheet" href="../highlight.css">
        <link rel="stylesheet" href="../tomorrow-night.css">
        <link rel="stylesheet" href="../ayu-highlight.css">

        <!-- Custom theme stylesheets -->

    </head>
    <body>
    <div id="body-container">
        <!-- Provide site root to javascript -->
        <script>
            var path_to_root = "../";
            var default_theme = window.matchMedia("(prefers-color-scheme: dark)").matches ? "navy" : "light";
        </script>

        <!-- Work around some values being stored in localStorage wrapped in quotes -->
        <script>
            try {
                var theme = localStorage.getItem('mdbook-theme');
                var sidebar = localStorage.getItem('mdbook-sidebar');

                if (theme.startsWith('"') && theme.endsWith('"')) {
                    localStorage.setItem('mdbook-theme', theme.slice(1, theme.length - 1));
                }

                if (sidebar.startsWith('"') && sidebar.endsWith('"')) {
                    localStorage.setItem('mdbook-sidebar', sidebar.slice(1, sidebar.length - 1));
                }
            } catch (e) { }
        </script>

        <!-- Set the theme before any content is loaded, prevents flash -->
        <script>
            var theme;
            try { theme = localStorage.getItem('mdbook-theme'); } catch(e) { }
            if (theme === null || theme === undefined) { theme = default_theme; }
            var html = document.querySelector('html');
            html.classList.remove('no-js')
            html.classList.remove('light')
            html.classList.add(theme);
            html.classList.add('js');
        </script>

        <!-- Hide / unhide sidebar before it is displayed -->
        <script>
            var html = document.querySelector('html');
            var sidebar = null;
            if (document.body.clientWidth >= 1080) {
                try { sidebar = localStorage.getItem('mdbook-sidebar'); } catch(e) { }
                sidebar = sidebar || 'visible';
            } else {
                sidebar = 'hidden';
            }
            html.classList.remove('sidebar-visible');
            html.classList.add("sidebar-" + sidebar);
        </script>

        <nav id="sidebar" class="sidebar" aria-label="Table of contents">
            <div class="sidebar-scrollbox">
                <ol class="chapter"><li class="chapter-item expanded affix "><a href="../en/introduction.html">Introduction</a></li><li class="chapter-item expanded affix "><a href="../en/updates.html">What's New?</a></li><li class="chapter-item expanded affix "><a href="../en/dependents.html">Dependents & News</a></li><li class="chapter-item expanded affix "><li class="part-title">User Guide</li><li class="chapter-item expanded "><a href="../en/installation.html">Installation</a></li><li class="chapter-item expanded "><a href="../en/quick_start_guide.html">Quick Start Guide</a></li><li class="chapter-item expanded affix "><li class="part-title">Manual</li><li class="chapter-item expanded "><a href="../en/alphafold.html">gget alphafold</a></li><li class="chapter-item expanded "><a href="../en/archs4.html">gget archs4</a></li><li class="chapter-item expanded "><a href="../en/bgee.html">gget bgee</a></li><li class="chapter-item expanded "><a href="../en/blast.html">gget blast</a></li><li class="chapter-item expanded "><a href="../en/blat.html">gget blat</a></li><li class="chapter-item expanded "><a href="../en/cbio.html">gget cbio</a></li><li class="chapter-item expanded "><a href="../en/cellxgene.html">gget cellxgene</a></li><li class="chapter-item expanded "><a href="../en/cosmic.html">gget cosmic</a></li><li class="chapter-item expanded "><a href="../en/diamond.html">gget diamond</a></li><li class="chapter-item expanded "><a href="../en/elm.html">gget elm</a></li><li class="chapter-item expanded "><a href="../en/enrichr.html">gget enrichr</a></li><li class="chapter-item expanded "><a href="../en/gpt.html">gget gpt</a></li><li class="chapter-item expanded "><a href="../en/info.html">gget info</a></li><li class="chapter-item expanded "><a href="../en/muscle.html">gget muscle</a></li><li class="chapter-item expanded "><a href="../en/mutate.html" class="active">gget mutate</a></li><li class="chapter-item expanded "><a href="../en/opentargets.html">gget opentargets</a></li><li class="chapter-item expanded "><a href="../en/pdb.html">gget pdb</a></li><li class="chapter-item expanded "><a href="../en/ref.html">gget ref</a></li><li class="chapter-item expanded "><a href="../en/search.html">gget search</a></li><li class="chapter-item expanded "><a href="../en/setup.html">gget setup</a></li><li class="chapter-item expanded "><a href="../en/seq.html">gget seq</a></li><li class="spacer"></li><li class="chapter-item expanded "><a href="../en/contributing.html">Contributing Guide</a></li><li class="chapter-item expanded "><a href="../en/code_of_conduct.html">Code of Conduct</a></li><li class="chapter-item expanded "><a href="../en/cite.html">Citation Guidelines</a></li><li class="spacer"></li><li class="chapter-item expanded affix "><li class="part-title">Español</li><li class="chapter-item expanded "><a href="../es/introduction.html">Introdución</a></li><li class="chapter-item expanded "><a href="../es/updates.html">¡Lo más reciente!</a></li><li class="chapter-item expanded "><a href="../es/dependents.html">Dependientes y Noticias</a></li><li class="chapter-item expanded affix "><li class="part-title">Guía del usario</li><li class="chapter-item expanded "><a href="../es/installation.html">Instalación</a></li><li class="chapter-item expanded "><a href="../es/quick_start_guide.html">Guía de inicio rápido</a></li><li class="chapter-item expanded affix "><li class="part-title">Manuál</li><li class="chapter-item expanded "><a href="../es/alphafold.html">gget alphafold</a></li><li class="chapter-item expanded "><a href="../es/archs4.html">gget archs4</a></li><li class="chapter-item expanded "><a href="../es/bgee.html">gget bgee</a></li><li class="chapter-item expanded "><a href="../es/blast.html">gget blast</a></li><li class="chapter-item expanded "><a href="../es/blat.html">gget blat</a></li><li class="chapter-item expanded "><a href="../es/cbio.html">gget cbio</a></li><li class="chapter-item expanded "><a href="../es/cellxgene.html">gget cellxgene</a></li><li class="chapter-item expanded "><a href="../es/cosmic.html">gget cosmic</a></li><li class="chapter-item expanded "><a href="../es/diamond.html">gget diamond</a></li><li class="chapter-item expanded "><a href="../es/elm.html">gget elm</a></li><li class="chapter-item expanded "><a href="../es/enrichr.html">gget enrichr</a></li><li class="chapter-item expanded "><a href="../es/gpt.html">gget gpt</a></li><li class="chapter-item expanded "><a href="../es/info.html">gget info</a></li><li class="chapter-item expanded "><a href="../es/muscle.html">gget muscle</a></li><li class="chapter-item expanded "><a href="../es/mutate.html">gget mutate</a></li><li class="chapter-item expanded "><a href="../es/opentargets.html">gget opentargets</a></li><li class="chapter-item expanded "><a href="../es/pdb.html">gget pdb</a></li><li class="chapter-item expanded "><a href="../es/ref.html">gget ref</a></li><li class="chapter-item expanded "><a href="../es/search.html">gget search</a></li><li class="chapter-item expanded "><a href="../es/setup.html">gget setup</a></li><li class="chapter-item expanded "><a href="../es/seq.html">gget seq</a></li><li class="spacer"></li><li class="chapter-item expanded "><a href="../es/contributing.html">Guía de contribución</a></li><li class="chapter-item expanded "><a href="../es/code_of_conduct.html">Codigo de conducto</a></li><li class="chapter-item expanded "><a href="../es/cite.html">Cómo citar</a></li></ol>
            </div>
            <div id="sidebar-resize-handle" class="sidebar-resize-handle"></div>
        </nav>

        <!-- Track and set sidebar scroll position -->
        <script>
            var sidebarScrollbox = document.querySelector('#sidebar .sidebar-scrollbox');
            sidebarScrollbox.addEventListener('click', function(e) {
                if (e.target.tagName === 'A') {
                    sessionStorage.setItem('sidebar-scroll', sidebarScrollbox.scrollTop);
                }
            }, { passive: true });
            var sidebarScrollTop = sessionStorage.getItem('sidebar-scroll');
            sessionStorage.removeItem('sidebar-scroll');
            if (sidebarScrollTop) {
                // preserve sidebar scroll position when navigating via links within sidebar
                sidebarScrollbox.scrollTop = sidebarScrollTop;
            } else {
                // scroll sidebar to current active section when navigating via "next/previous chapter" buttons
                var activeSection = document.querySelector('#sidebar .active');
                if (activeSection) {
                    activeSection.scrollIntoView({ block: 'center' });
                }
            }
        </script>

        <div id="page-wrapper" class="page-wrapper">

            <div class="page">
                                <div id="menu-bar-hover-placeholder"></div>
                <div id="menu-bar" class="menu-bar sticky">
                    <div class="left-buttons">
                        <button id="sidebar-toggle" class="icon-button" type="button" title="Toggle Table of Contents" aria-label="Toggle Table of Contents" aria-controls="sidebar">
                            <i class="fa fa-bars"></i>
                        </button>
                        <button id="theme-toggle" class="icon-button" type="button" title="Change theme" aria-label="Change theme" aria-haspopup="true" aria-expanded="false" aria-controls="theme-list">
                            <i class="fa fa-paint-brush"></i>
                        </button>
                        <ul id="theme-list" class="theme-popup" aria-label="Themes" role="menu">
                            <li role="none"><button role="menuitem" class="theme" id="light">Light</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="rust">Rust</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="coal">Coal</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="navy">Navy</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="ayu">Ayu</button></li>
                        </ul>
                        <button id="search-toggle" class="icon-button" type="button" title="Search. (Shortkey: s)" aria-label="Toggle Searchbar" aria-expanded="false" aria-keyshortcuts="S" aria-controls="searchbar">
                            <i class="fa fa-search"></i>
                        </button>
                    </div>

                    <h1 class="menu-title">gget</h1>

                    <div class="right-buttons">
                        <a href="../print.html" title="Print this book" aria-label="Print this book">
                            <i id="print-button" class="fa fa-print"></i>
                        </a>

                    </div>
                </div>

                <div id="search-wrapper" class="hidden">
                    <form id="searchbar-outer" class="searchbar-outer">
                        <input type="search" id="searchbar" name="searchbar" placeholder="Search this book ..." aria-controls="searchresults-outer" aria-describedby="searchresults-header">
                    </form>
                    <div id="searchresults-outer" class="searchresults-outer hidden">
                        <div id="searchresults-header" class="searchresults-header"></div>
                        <ul id="searchresults">
                        </ul>
                    </div>
                </div>

                <!-- Apply ARIA attributes after the sidebar and the sidebar toggle button are added to the DOM -->
                <script>
                    document.getElementById('sidebar-toggle').setAttribute('aria-expanded', sidebar === 'visible');
                    document.getElementById('sidebar').setAttribute('aria-hidden', sidebar !== 'visible');
                    Array.from(document.querySelectorAll('#sidebar a')).forEach(function(link) {
                        link.setAttribute('tabIndex', sidebar === 'visible' ? 0 : -1);
                    });
                </script>

                <div id="content" class="content">
                    <main>
                        <blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python.  The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h1 id="gget-mutate-"><a class="header" href="#gget-mutate-">gget mutate 🧟</a></h1>
<p>Takes in nucleotide sequences and mutations (in <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1867422/">standard mutation annotation</a> and returns mutated versions of the input sequences according to the provided mutations.<br />
Return format: Saves mutated sequences in FASTA format (or returns a list containing the mutated sequences if <code>out=None</code>).</p>
<p>This module was written by <a href="https://github.com/josephrich98">Joseph Rich</a>.</p>
<p><strong>Positional argument</strong><br />
<code>sequences</code><br />
Path to the FASTA file containing the sequences to be mutated, e.g., 'path/to/seqs.fa'.<br />
Sequence identifiers following the '&gt;' character must correspond to the identifiers in the seq_ID column of <code>mutations</code>.</p>
<p>Example format of the FASTA file:</p>
<pre><code>&gt;seq1 (or ENSG00000106443)  
ACTGCGATAGACT  
&gt;seq2  
AGATCGCTAG
</code></pre>
<p>Alternatively: Input sequence(s) as a string or list, e.g. 'AGCTAGCT'.</p>
<p>NOTE: Only the letters until the first space or dot will be used as sequence identifiers - Version numbers of Ensembl IDs will be ignored.<br />
NOTE: When the <code>sequences</code> input is a genome fasta file, also see the <code>gtf</code> argument below.</p>
<p><strong>Required arguments</strong><br />
<code>-m</code> <code>--mutations</code><br />
Path to the csv or tsv file (e.g., 'path/to/mutations.csv') or data frame (DataFrame object) containing information about the mutations in the following format (the 'notes' and 'mut_ID' columns are optional):</p>
<div class="table-wrapper"><table><thead><tr><th>mutation</th><th>mut_ID</th><th>seq_ID</th><th>notes</th></tr></thead><tbody>
<tr><td>c.2C&gt;T</td><td>mut1</td><td>seq1</td><td>-&gt; Apply mutation 1 to sequence 1</td></tr>
<tr><td>c.9_13inv</td><td>mut2</td><td>seq2</td><td>-&gt; Apply mutation 2 to sequence 2</td></tr>
<tr><td>c.9_13inv</td><td>mut2</td><td>seq4</td><td>-&gt; Apply mutation 2 to sequence 4</td></tr>
<tr><td>c.9_13delinsAAT</td><td>mut3</td><td>seq4</td><td>-&gt; Apply mutation 3 to sequence 4</td></tr>
<tr><td>...</td><td>...</td><td>...</td><td></td></tr>
</tbody></table>
</div>
<p>'mutation' = Column containing the mutations to be performed written in standard mutation annotation<br />
'mut_ID' = Column containing the identifier for each mutation<br />
'seq_ID' = Column containing the identifiers of the sequences to be mutated (must correspond to the string following the '&gt;' character in the 'sequences' FASTA file; do NOT include spaces or dots)</p>
<p>Alternatively: Input mutation(s) as a string or list, e.g., 'c.2C&gt;T'.<br />
If a list is provided, the number of mutations must equal the number of input sequences.</p>
<p>For use from the terminal (bash): Enclose individual mutation annotations in quotation marks to prevent parsing errors.</p>
<p><strong>Optional input-related arguments</strong><br />
<code>-mc</code> <code>--mut_column</code><br />
Name of the column containing the mutations to be performed in <code>mutations</code>. Default: 'mutation'.</p>
<p><code>-sic</code> <code>--seq_id_column</code><br />
Name of the column containing the IDs of the sequences to be mutated in <code>mutations</code>. Default: 'seq_ID'.</p>
<p><code>-mic</code> <code>--mut_id_column</code><br />
Name of the column containing the IDs of each mutation in <code>mutations</code>. Default: Same as <code>mut_column</code>.</p>
<p><code>-gtf</code> <code>--gtf</code><br />
Path to a .gtf file. When providing a genome fasta file as input for 'sequences', you can provide a .gtf file here and the input sequences will be defined according to the transcript boundaries, e.g. 'path/to/genome_annotation.gtf'. Default: None</p>
<p><code>-gtic</code> <code>--gtf_transcript_id_column</code><br />
Column name in the input <code>mutations</code> file containing the transcript ID. In this case, column <code>seq_id_column</code> should contain the chromosome number.<br />
Required when <code>gtf</code> is provided. Default: None</p>
<p><strong>Optional mutant sequence generation/filtering arguments</strong><br />
<code>-k</code> <code>--k</code><br />
Length of sequences flanking the mutation. Default: 30.<br />
If k &gt; total length of the sequence, the entire sequence will be kept.</p>
<p><code>-msl</code> <code>--min_seq_len</code><br />
Minimum length of the mutant output sequence, e.g. 100. Mutant sequences smaller than this will be dropped. Default: None</p>
<p><code>-ma</code> <code>--max_ambiguous</code><br />
Maximum number of 'N' (or 'n') characters allowed in the output sequence, e.g. 10. Default: None (no ambiguous character filter will be applied)</p>
<p><strong>Optional mutant sequence generation/filtering flags</strong><br />
<code>-ofr</code> <code>--optimize_flanking_regions</code><br />
Removes nucleotides from either end of the mutant sequence to ensure (when possible) that the mutant sequence does not contain any k-mers also found in the wildtype/input sequence. </p>
<p><code>-rswk</code> <code>--remove_seqs_with_wt_kmers</code><br />
Removes output sequences where at least one k-mer is also present in the wildtype/input sequence in the same region.<br />
When used with <code>--optimize_flanking_regions</code>, only sequences for which a wildtype k-mer is still present after optimization will be removed.</p>
<p><code>-mio</code> <code>--merge_identical_off</code><br />
Do not merge identical mutant sequences in the output (by default, identical sequences will be merged by concatenating the sequence headers for all identical sequences).</p>
<p><strong>Optional arguments to generate additional output</strong><br />
This output is activated using the  <code>--update_df</code> flag and will be stored in a copy of the <code>mutations</code> DataFrame.</p>
<p><code>-udf_o</code> <code>--update_df_out</code><br />
Path to output csv file containing the updated DataFrame, e.g. 'path/to/mutations_updated.csv'. Only valid when used with <code>--update_df</code>.<br />
Default: None -&gt; the new csv file will be saved in the same directory as the <code>mutations</code> DataFrame with appendix '_updated'</p>
<p><code>-ts</code> <code>--translate_start</code><br />
(int or str) The position in the input nucleotide sequence to start translating, e.g. 5. If a string is provided, it should correspond to a column name in <code>mutations</code> containing the open reading frame start positions for each sequence/mutation. Only valid when used with <code>--translate</code>.<br />
Default: translates from the beginning of each sequence</p>
<p><code>-te</code> <code>--translate_end</code><br />
(int or str) The position in the input nucleotide sequence to end translating, e.g. 35. If a string is provided, it should correspond to a column name in <code>mutations</code> containing the open reading frame end positions for each sequence/mutation. Only valid when used with <code>--translate</code>.<br />
Default: translates until the end of each sequence</p>
<p><strong>Optional flags to modify additional output</strong><br />
<code>-udf</code> <code>--update_df</code><br />
Updates the input <code>mutations</code> DataFrame to include additional columns with the mutation type, wildtype nucleotide sequence, and mutant nucleotide sequence (only valid if <code>mutations</code> is a .csv or .tsv file).</p>
<p><code>-sfs</code> <code>--store_full_sequences</code><br />
Includes the complete wildtype and mutant sequences in the updated <code>mutations</code> DataFrame (not just the sub-sequence with k-length flanks). Only valid when used with <code>--update_df</code>.</p>
<p><code>-tr</code> <code>--translate</code><br />
Adds additional columns to the updated <code>mutations</code> DataFrame containing the wildtype and mutant amino acid sequences. Only valid when used with <code>--store_full_sequences</code>.</p>
<p><strong>Optional general arguments</strong><br />
<code>-o</code> <code>--out</code><br />
Path to output FASTA file containing the mutated sequences, e.g., 'path/to/output_fasta.fa'.<br />
Default: None -&gt; returns a list of the mutated sequences to standard out.<br />
The identifiers (following the '&gt;') of the mutated sequences in the output FASTA will be '&gt;[seq_ID]_[mut_ID]'. </p>
<p><strong>Optional general flags</strong><br />
<code>-q</code> <code>--quiet</code><br />
Command-line only. Prevents progress information from being displayed.<br />
Python: Use <code>verbose=False</code> to prevent progress information from being displayed.</p>
<h3 id="examples"><a class="header" href="#examples">Examples</a></h3>
<pre><code class="language-bash">gget mutate ATCGCTAAGCT -m 'c.4G&gt;T'
</code></pre>
<pre><code class="language-python"># Python
gget.mutate(&quot;ATCGCTAAGCT&quot;, &quot;c.4G&gt;T&quot;)
</code></pre>
<p>→ Returns ATCTCTAAGCT.</p>
<p><br/><br/></p>
<p><strong>List of sequences with a mutation for each sequence provided in a list:</strong></p>
<pre><code class="language-bash">gget mutate ATCGCTAAGCT TAGCTA -m 'c.4G&gt;T' 'c.1_3inv' -o mut_fasta.fa
</code></pre>
<pre><code class="language-python"># Python
gget.mutate([&quot;ATCGCTAAGCT&quot;, &quot;TAGCTA&quot;], [&quot;c.4G&gt;T&quot;, &quot;c.1_3inv&quot;], out=&quot;mut_fasta.fa&quot;)
</code></pre>
<p>→ Saves 'mut_fasta.fa' file containing: </p>
<pre><code>&gt;seq1_mut1  
ATCTCTAAGCT  
&gt;seq2_mut2  
GATCTA
</code></pre>
<p><br/><br/></p>
<p><strong>One mutation applied to several sequences with adjusted <code>k</code>:</strong></p>
<pre><code class="language-bash">gget mutate ATCGCTAAGCT TAGCTA -m 'c.1_3inv' -k 3
</code></pre>
<pre><code class="language-python"># Python
gget.mutate([&quot;ATCGCTAAGCT&quot;, &quot;TAGCTA&quot;], &quot;c.1_3inv&quot;, k=3)
</code></pre>
<p>→ Returns ['CTAGCT', 'GATCTA'].</p>
<p><br/><br/></p>
<p><strong>Add mutations to an entire genome with extended output</strong><br />
Main input:</p>
<ul>
<li>mutation information as a <code>mutations</code> CSV (by having <code>seq_id_column</code> contain chromosome information, and <code>mut_column</code> contain mutation information with respect to genome coordinates)</li>
<li>the genome as the <code>sequences</code> file</li>
</ul>
<p>Since we are passing the path to a gtf file to the <code>gtf</code> argument, transcript boundaries will be respected (the genome will be split into transcripts). <code>gtf_transcript_id_column</code> specifies the name of the column in <code>mutations</code> containing the transcript IDs corresponding to the transcript IDs in the <code>gtf</code> file.</p>
<p>The <code>optimize_flanking_regions</code> argument maximizes the length of the resulting mutation-containing sequences while maintaining specificity (no wildtype k-mer will be retained).</p>
<p><code>update_df</code> activates the creation of a new CSV file with updated information about each input and output sequence. This new CSV file will be saved as <code>update_df_out</code>. Since <code>store_full_sequences</code> is activated, this new CSV file will not only contain the output sequences (restricted in size by flanking regiong of size <code>k</code>), but also the complete input and output sequences. This allows us to observe the mutation in the context of the entire sequence. Lastly, we are also adding the translated versions of the complete sequences by adding the with the <code>translate</code> flag, so we can observe how the resulting amino acid sequence is changed. The <code>translate_start</code> and <code>translate_end</code> arguments specify the names of the columns in <code>mutations</code> that contain the start and end positions of the open reading frame (start and end positions for translating the nucleotide sequence to an amino acid sequence), respectively.</p>
<pre><code class="language-bash">gget mutate \
  -m mutations_input.csv \
  -o mut_fasta.fa \
  -k 4 \
  -sic Chromosome \
  -mic Mutation \
  -gtf genome_annotation.gtf \
  -gtic Ensembl_Transcript_ID \
  -ofr \
  -update_df \
  -udf_o mutations_updated.csv \
  -sfs \
  -tr \
  -ts Translate_Start \
  -te Translate_End \
  genome_reference.fa
</code></pre>
<pre><code class="language-python"># Python
gget.mutate(
  sequences=&quot;genome_reference.fa&quot;,
  mutations=&quot;mutations_input.csv&quot;,
  out=&quot;mut_fasta.fa&quot;,
  k=4,
  seq_id_column=&quot;Chromosome&quot;,
  mut_column=&quot;Mutation&quot;,
  gtf=&quot;genome_annotation.gtf&quot;,
  gtf_transcript_id_column=&quot;Ensembl_Transcript_ID&quot;,
  optimize_flanking_regions=True,
  update_df=True,
  update_df_out=&quot;mutations_updated.csv&quot;,
  store_full_sequences=True,
  translate=True,
  translate_start=&quot;Translate_Start&quot;,
  translate_end=&quot;Translate_End&quot;
)
</code></pre>
<p>→ Takes in a genome fasta ('genome_reference.fa') and gtf file ('genome_annotation.gtf') (these can be downloaded using <a href="ref.html"><code>gget ref</code></a>) as well as a 'mutations_input.csv' file containing: </p>
<pre><code>| Chromosome | Mutation          | Ensembl_Transcript_ID  | Translate_Start | Translate_End |
|------------|-------------------|------------------------|-----------------|---------------|
| 1          | g.224411A&gt;C       | ENST00000193812        | 0               | 100           |
| 8          | g.25111del        | ENST00000174411        | 0               | 294           |
| X          | g.1011_1012insAA  | ENST00000421914        | 9               | 1211          |
</code></pre>
<p>→ Saves 'mut_fasta.fa' file containing: </p>
<pre><code>&gt;1:g.224411A&gt;C  
TGCTCTGCT  
&gt;8:g.25111del  
GAGTCGAT
&gt;X:g.1011_1012insAA
TTAGAACTT
</code></pre>
<p>→ Saves 'mutations_updated.csv' file containing: </p>
<pre><code>
| Chromosome | Mutation          | Ensembl_Transcript_ID  | mutation_type | wt_sequence | mutant_sequence | wt_sequence_full  | mutant_sequence_full | wt_sequence_aa_full | mutant_sequence_aa_full |
|------------|-------------------|------------------------|---------------|-------------|-----------------|-------------------|----------------------|---------------------|-------------------------|
| 1          | g.224411A&gt;C       | ENSMUST00000193812     | Substitution  | TGCTATGCT   | TGCTCTGCT       | ...TGCTATGCT...   | ...TGCTCTGCT...      | ...CYA...           | ...CSA...               |
| 8          | g.25111del        | ENST00000174411        | Deletion      | GAGTCCGAT   | GAGTCGAT        | ...GAGTCCGAT...   | ...GAGTCGAT...       | ...ESD...           | ...ES...                |
| X          | g.1011_1012insAA  | ENST00000421914        | Insertion     | TTAGCTT     | TTAGAACTT       | ...TTAGCTT...     | ...TTAGAACTT...      | ...A...             | ...EL...                |

</code></pre>
<h1 id="references"><a class="header" href="#references">References</a></h1>
<p>If you use <code>gget mutate</code> in a publication, please cite the following articles:</p>
<ul>
<li>Luebbert, L., &amp; Pachter, L. (2023). Efficient querying of genomic reference databases with gget. Bioinformatics. <a href="https://doi.org/10.1093/bioinformatics/btac836">https://doi.org/10.1093/bioinformatics/btac836</a></li>
</ul>

                    </main>

                    <nav class="nav-wrapper" aria-label="Page navigation">
                        <!-- Mobile navigation buttons -->
                            <a rel="prev" href="../en/muscle.html" class="mobile-nav-chapters previous" title="Previous chapter" aria-label="Previous chapter" aria-keyshortcuts="Left">
                                <i class="fa fa-angle-left"></i>
                            </a>

                            <a rel="next" href="../en/opentargets.html" class="mobile-nav-chapters next" title="Next chapter" aria-label="Next chapter" aria-keyshortcuts="Right">
                                <i class="fa fa-angle-right"></i>
                            </a>

                        <div style="clear: both"></div>
                    </nav>
                </div>
            </div>

            <nav class="nav-wide-wrapper" aria-label="Page navigation">
                    <a rel="prev" href="../en/muscle.html" class="nav-chapters previous" title="Previous chapter" aria-label="Previous chapter" aria-keyshortcuts="Left">
                        <i class="fa fa-angle-left"></i>
                    </a>

                    <a rel="next" href="../en/opentargets.html" class="nav-chapters next" title="Next chapter" aria-label="Next chapter" aria-keyshortcuts="Right">
                        <i class="fa fa-angle-right"></i>
                    </a>
            </nav>

        </div>




        <script>
            window.playground_copyable = true;
        </script>


        <script src="../elasticlunr.min.js"></script>
        <script src="../mark.min.js"></script>
        <script src="../searcher.js"></script>

        <script src="../clipboard.min.js"></script>
        <script src="../highlight.js"></script>
        <script src="../book.js"></script>

        <!-- Custom JS scripts -->


    </div>
    </body>
</html>
